gammapy/scripts/spectrum_pipe.py

```diff
@@ -43,11 +43,17 @@ class SpectrumAnalysisIACT(object):
         Observations to analyse
     config : dict
         Config dict
+    nsigma_ul_threshold : float
```

Comment:
I think these parameters could (and should) be passed as part of the ``config`` dict. The idea is, that this config dict at one point can be read from a file.

Reply:
This is a good point. In Gammpy functions, I do not see a clear and agreed method to pass arguments. Currently, it uses either arguments or a config dict.
What should be you use? According to the choice, many classes should be updated if only 1 way is chosen...
I open then the debate;)

Reply:
You are right, that there a many things in Gammapy where a coherent structure is missing and a debate is needed at some point. Maybe it's a little out of scope for this PR? At the moment almost all classes take python objects as input in the __init__ method. Some classes have (or at least had at some point) a ``from_config`` method to initialize them with a config dict. I added the  ``SpectrumAnalysisIACT`` as a high level class to be used only with a config dict.

No matter which API we choose in the end, it would be nice also now to stay consistent at least on the scale of one class, don't you think? If there is a reason why the new parameter don't fit in the  config dict, we can of course try to adapt the ``SpectrumAnalysisIACT`` once we have a better idea of the broader API of gammapy.

Reply:
Your point is correct. For this class, a dictionary is used and it is better at this stage to follow this scheme.
I have adapted the code.